### PR TITLE
feat(cli): `compile --output`

### PR DESCRIPTION
**Pull request by tsuf239**
## Checklist
fixes: https://github.com/winglang/wing/issues/6340

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: apps/wing/src/commands/compile.ts [hunk 1]
```diff
@@ -49,6 +49,11 @@ export interface CompileOptions {
    * @default "./target"
    */
   readonly targetDir?: string;
+  /**
+   * The overrides the location to save the compilation output
+   * @default "./target"
```
Comment (eladb) [inline apps/wing/src/commands/compile.ts:54]: The default is `./target/<entrypoint>.<target>` -- we should make sure this overrides the entire path and not just `./target`.

What I mean is that this should behave like this:

```sh
$ wing compile -t tf-aws main.w -o my-output
$ cd my-output
$ terraform init
$ terraform apply
```

**Review by eladb — COMMENTED** [review]
please change the PR title to match the issue title

File: apps/wing/src/commands/compile.ts (hunk 1 above)
Comment (tsuf239) [reply apps/wing/src/commands/compile.ts]: it's only the comment that was wrong :)

File: libs/wingcompiler/src/compile.test.ts [hunk 2]
```diff
@@ -12,60 +12,72 @@ export async function generateTmpDir() {
...
+    expect(stats.isDirectory()).toBeTruthy();
+    expect(artifactDir).toContain(targetDir);
+    expect(basename(artifactDir)).toEqual("enums.test.wsim");
+  });
 
-      const stats = await stat(artifactDir);
-      expect(stats.isDirectory()).toBeTruthy();
-      expect(artifactDir).toContain(targetDir);
-      expect(basename(artifactDir)).toEqual("enums.test.wsim")
+  test("should be able to override the target directory", async () => {
```
Comment (tsuf239) [inline libs/wingcompiler/src/compile.test.ts:70]: this is the only new test (rest is formatting)

File: libs/wingcompiler/src/compile.ts [hunk 3]
```diff
@@ -63,6 +63,9 @@ export interface CompileOptions {
 
   // target directory for the output files
   readonly targetDir?: string;
+
+  // overrides the target directory for the output files
+  readonly output?: string;
```
Comment (eladb) [inline libs/wingcompiler/src/compile.ts:68]:
```suggestion
  readonly outputOverride?: string;
```

Add description of the default behavior to docstrings

Comment (tsuf239) [reply libs/wingcompiler/src/compile.ts:68]: I think there is a value in keeping it aligned to the cli (we pass all the options with a spread operator) 
looked at the cli tool and there is no way to change the variable :(

**Pull request merged**